### ISSUE TITLE
Remove subject linebreaks and tabs in thread_index

### ISSUE THREAD

**Issue by mxmehl** [astroidmail/astroid#361]
Some clients make linebreaks in the header's subject line when the title is long. For example:

```
Subject: [Feedback until 26 June]Online platform liability - project
	collaboration
```
In Astroid's thread_index, these line breaks or the tabs/spaces added cause an ugly space inside the subject line: http://i.imgur.com/KxVg8zi.png

Interestingly, in thrad_view the space isn't shown between the two words.

**Comment by gauteh** [comment]
Max writes on juni 22, 2017 13:23:
> Interestingly, in thrad_view the space isn't shown between the two words.

Is it present in the HTML source?

**Comment by mxmehl** [comment]
>> Interestingly, in thrad_view the space isn't shown between the two words.
>
> Is it present in the HTML source?

Yes, it seems to contain a tab, at least when I copy it to a text editor:

```
<div class="value">[Feedback until 26 June]Online platform liability - project	collaboration</div>
```

**Comment by gauteh** [comment]
Max writes on juni 22, 2017 13:30:
>>> Interestingly, in thrad_view the space isn't shown between the two words.
>>
>> Is it present in the HTML source?
> 
> Yes, it seems to contain a tab, at least when I copy it to a text editor:
> 
> ```
> <div class="value">[Feedback until 26 June]Online platform liability - project	collaboration</div>
> ```

In that case I think it is gmime's reponsibility. I'm guessing the 
e-mail clients are not following the spec's?

**Comment by mxmehl** [comment]
Am 22.06.2017 um 13:33 schrieb Gaute Hope:
> In that case I think it is gmime's reponsibility. I'm guessing the
> e-mail clients are not following the spec's?

Dunno but it seems to happen quite often, and I guess not caused by
crappy Outlook but Free Software MUAs.

The line break seems to be normal but the addition of a tab is causing
issues.

**Comment by gauteh** [comment]
@mxmehl did you have a chance to test with those gmime functions?

**Comment by mxmehl** [comment]
No, not yet unfortunately...
